Tests/MapboxMapsTests/Foundation/StyleTests.swift

```diff
@@ -344,4 +344,44 @@ final class StyleTests: XCTestCase {
             bounds: CoordinateBounds(southwest: .random(), northeast: .random()))
         )
     }
+
+    func testStyleCanUpdateLayer() throws {
```

Comment:
Add tests that check that:
* nilling out unmodified properties works correctly
* updating a property value that already has a value works correctly

Reply:
done!